cpp/log/strict_consistent_store_test.cc

```diff
@@ -0,0 +1,134 @@
+#include "log/strict_consistent_store.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "log/logged_certificate.h"
+#include "log/mock_consistent_store.h"
+#include "util/mock_masterelection.h"
+#include "util/status.h"
+#include "util/statusor.h"
+#include "util/testing.h"
+#include "util/util.h"
+
+DECLARE_int32(node_state_ttl_seconds);
+
+namespace cert_trans {
+
+using testing::_;
+using testing::NiceMock;
+using testing::Return;
+using util::Status;
+using util::StatusOr;
+
+
+class StrictConsistentStoreTest : public ::testing::TestWithParam<bool> {
+ public:
+  StrictConsistentStoreTest()
+      : peer_(new NiceMock<MockConsistentStore<LoggedCertificate>>()),
+        strict_store_(
+            new StrictConsistentStore<LoggedCertificate>(&election_, peer_)) {
+    ON_CALL(election_, IsMaster()).WillByDefault(Return(IsMaster()));
+  }
+
+ protected:
+  bool IsMaster() const {
+    return GetParam();
+  }
+
+  NiceMock<MockMasterElection> election_;
+  // strict_store_ takes ownership of this:
+  NiceMock<MockConsistentStore<LoggedCertificate>>* peer_;
+  std::unique_ptr<StrictConsistentStore<LoggedCertificate>> strict_store_;
```

Comment:
D'oh, you're right of course - this was a hang-over from when I was creating a new one in `SetUp()` each time while trying to narrow down the leaked Mock (which, of course, was actually the lack of the `virtual` d'tor.)
Unpointered.